parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -33,6 +33,8 @@ if [[ "${TARGET_ENVIRONMENT,,}" == "${AZURE_STACK_ENV}"  ]]; then
     config_script_custom_cloud=/opt/azure/containers/provision_configs_custom_cloud.sh
     wait_for_file 3600 1 $config_script_custom_cloud || exit $ERR_FILE_WATCH_TIMEOUT
     source $config_script_custom_cloud
+    KUBERNETES_VERSION_NO_SUFFIX="${KUBERNETES_VERSION}"
```

Comment:
instead of changing `KUBERNETES_VERSION`, let's make a new var `AZURE_STACK_KUBERNETES_VERSION` and use that in the `extractHyperkube` function if and only if the target env is Azure Stack. Are there any other places where the Azure Stack prefix is needed besides `extractHyperkube`?

Reply:
extractHyperkube, installKubeletAndKubectl, cleanUpContainerImages

Reply:
let me know please if this my last reply changes the new approach in someway

Reply:
I'm trying to think of the best way to do this. We need to put all these changes in  `AZURE_STACK_ENV` flows as much as possible so we don't break non-Azure-Stack scenarios.

Reply:
np, let me know please the approach you guys prefer
- alter functions vs alter code consuming functions?
- global variables vs function parameters?